### PR TITLE
pbs: new virtual package

### PR DESCRIPTION
**Pull request by sethrj**
Some of our clusters have an older installation of libtorque and tm.h that are *not* from OpenPBS. (It's apparently TORQUE 3.0.2, possibly from 2008ish.) Using the current openpbs dependency for openmpi causes concretization errors due to restrictions on older python and hwloc requirements that don't apply, even with an external non-buildable installation. The new 'torque' bundle package allows users to point to that external installation without problems.

@skosukhin It looks like you added the OpenPBS package and the explicit dependency in `openmpi schedulers=tm`, so please let me know if I'm missing something. Thanks!

### CODE REVIEW

**Comment by hppritcha** [comment]
I would like to see these changes tested on a system using OpenPBS.  I don't have access to such a system.

**Comment by sethrj** [comment]
I'm testing the change on the torque system now and will post confirmation. Before #16758, the `tm` scheduler option automatically worked by pulling in the implicit system dependency on our external libtorque.

**Comment by sethrj** [comment]
Yes, with this change, OpenMPI successfully installs with PBS support:
```
--- MCA component ess:tm (m4 configuration macro)
checking for MCA component ess:tm compile mode... static
checking --with-tm value... sanity check ok (/opt/pbs-3.0.2)
checking for pbs-config... /opt/pbs-3.0.2/bin/pbs-config
checking tm.h usability... yes
checking tm.h presence... yes
checking for tm.h... yes
checking for tm_finalize... yes
checking if MCA component ess:tm can compile... yes
```

**Comment by hppritcha** [comment]
Then I am okay with this change to the openmpi spackage,

**Comment by skosukhin** [comment]
@sethrj I don't think that `torque` should be a `BundlePackage`. I would implement it like other packages of this kind. For example: https://github.com/spack/spack/blob/a7b6149cc02713de55cb5719b9dab2dfc0476303/var/spack/repos/builtin/packages/mxm/package.py#L9-L32

**Comment by sethrj** [comment]
Thanks @skosukhin , I've taken up the suggestion. The `apple-libuuid` was the first package that came to my mind as uninstallable, but I've made it a regular `Package` as you suggested.

**Comment by sethrj** [comment]
More CI failures...

File: var/spack/repos/builtin/packages/torque/package.py
Comment (skosukhin) [inline var/spack/repos/builtin/packages/torque/package.py]:
```suggestion
    """TORQUE (Terascale Open-source Resource and QUEue Manager) is an open
    source project based on the original PBS resource manager developed by NASA,
    LLNL, and MRJ."""
```

**Review by skosukhin — DISMISSED** [review]

**Comment by sethrj** [comment]
Thanks @skosukhin !

**Comment by sethrj** [comment]
@spackbot re-run ci

**Comment by sethrj** [comment]
@spackbot re-run pipeline

**Comment by sethrj** [comment]
Rebased because spackbot didn't actually rerun the failed test it seems...

**Comment by sethrj** [comment]
@balay Could you give this a quick check? I already got approval from @skosukhin and @hppritcha but had to rebase due to a glitch in the test server. Thanks!

File: var/spack/repos/builtin/packages/torque/package.py
Comment (balay) [inline var/spack/repos/builtin/packages/torque/package.py:16]: Perhaps add in `maintainers`?

Comment (sethrj) [reply var/spack/repos/builtin/packages/torque/package.py:16]: Done

**Review by balay — APPROVED** [review]
Changes look fine to me!

**Pull request merged**
